release/cli/cli.go

```diff
@@ -0,0 +1,204 @@
+package cli
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"text/template"
+
+	"github.com/google/go-github/v39/github"
+	"github.com/rancher/ecm-distro-tools/cmd/release/config"
+	ecmConfig "github.com/rancher/ecm-distro-tools/cmd/release/config"
+	ecmExec "github.com/rancher/ecm-distro-tools/exec"
+	"github.com/rancher/ecm-distro-tools/release"
+	"github.com/rancher/ecm-distro-tools/repository"
+	"golang.org/x/mod/semver"
+)
+
+const (
+	cliOrg           = "rancher"
+	cliRepo          = "cli"
+	cliImagesBaseURL = "https://github.com/" + cliOrg + "/" + cliRepo + "/releases"
+)
+
+// CreateRelease will create a new tag and a new release with given params.
+func CreateRelease(ctx context.Context, client *github.Client, r *ecmConfig.CLIRelease, opts *repository.CreateReleaseOpts, rc bool, releaseType string) error {
+	if !semver.IsValid(opts.Tag) {
+		return errors.New("tag isn't a valid semver: " + opts.Tag)
+	}
+
+	latestPreRelease, err := release.LatestPreRelease(ctx, client, opts.Owner, opts.Repo, opts.Tag, releaseType)
+	if err != nil {
+		return err
+	}
+
+	opts.Name = opts.Tag
+	opts.Prerelease = true
+	opts.Draft = !rc
+	opts.ReleaseNotes = ""
+
+	if rc {
+		latestRCNumber := 1
+		if latestPreRelease != nil {
+			// v2.9.0-rc.N / -alpha.N
+			_, trimmedRCNumber, found := strings.Cut(*latestPreRelease, "-"+releaseType+".")
+			if !found {
+				return errors.New("failed to parse rc number from " + *latestPreRelease)
+			}
+			currentRCNumber, err := strconv.Atoi(trimmedRCNumber)
+			if err != nil {
+				return err
+			}
+			latestRCNumber = currentRCNumber + 1
+		}
+		opts.Tag = fmt.Sprintf("%s-%s.%d", opts.Tag, releaseType, latestRCNumber)
+	} else {
+		fmt.Printf("release.GenReleaseNotes(ctx, %s, %s, %s, %s, client)", opts.Owner, opts.Repo, opts.Branch, r.PreviousTag)
+		buff, err := release.GenReleaseNotes(ctx, opts.Owner, opts.Repo, opts.Branch, r.PreviousTag, client)
+		if err != nil {
+			return err
+		}
+		opts.ReleaseNotes = buff.String()
+	}
+
+	fmt.Printf("create release options: %+v\n", *opts)
```

Comment:
Was this for debug or expected to be output?

Reply:
expected output, the same is in other releases.